### PR TITLE
[CALCITE-3447] MutableScans with the same qualified name are not equivalent.

### PR DESCRIPTION
**Pull request by jinxing64**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/mutable/MutableScan.java [hunk 1]
```diff
@@ -36,11 +36,12 @@ public static MutableScan of(TableScan scan) {
   @Override public boolean equals(Object obj) {
     return obj == this
         || obj instanceof MutableScan
-        && rel.equals(((MutableScan) obj).rel);
+        && rel.getTable().getQualifiedName().equals(((MutableScan) obj).rel
```
Comment (DonnyZone) [inline core/src/main/java/org/apache/calcite/rel/mutable/MutableScan.java:39]: Can wet fix the `hashCode` function in `TableScan` directly?

Comment (jinxing64) [reply core/src/main/java/org/apache/calcite/rel/mutable/MutableScan.java:39]: Hi @DonnyZone  Thanks a lot for looking into this !
There's a discussion for overriding `equals()&hashCode()` in https://issues.apache.org/jira/browse/CALCITE-3167
The conclusion is to make `equals()&hashCode()` final in `AbstractRelNode ` and  have independent mechanisms to define the equality.

Comment (DonnyZone) [reply core/src/main/java/org/apache/calcite/rel/mutable/MutableScan.java:39]: @jinxing64 Thanks for clarification. LGTM!

**Comment by DonnyZone** [comment]
By the way, we'd better remove the period at the end of the commit message.

**Comment by jinxing64** [comment]
Sure ~ do now.

**Comment by jinxing64** [comment]
Thanks for merging ! @DonnyZone

**Pull request merged**